src/Bicep.Core/SemanticModel/NameBindingVisitor.cs

```diff
@@ -110,30 +131,61 @@ private Symbol ValidateFunctionFlags(Symbol symbol, TextSpan span)
             return symbol;
         }
 
-        private Symbol LookupSymbolByName(string name, TextSpan span)
+        private Symbol LookupSymbolByName(string name, TextSpan span, string? @namespace)
         {
-            if (this.declarations.TryGetValue(name, out var localSymbol))
+            NamespaceSymbol? FindNamespace(string name) =>
```

Comment:
Should we make `namespaces` into a `Dictionary` to speed up this lookup?

Reply:
There are two, so we actually didn't need to change it. O(2) = O(1) 😝 It's also fine as-is, so no need to change it back.

Reply:
Yeah, we might see the benefits once we add more namespaces.